### ISSUE TITLE
Checkout: Textarea fields do not inherit global text color styles

### ISSUE THREAD

**Issue by mikejolley** [woothemes/woocommerce#48898]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

This PR updated checkout inputs to pull in global styles:<br>[https://github.com/woocommerce/woocommerce/pull/46362](<https://github.com/woocommerce/woocommerce/pull/46362>)

Regular input placeholders are styled, but textarea placeholders are not.

### Expected behavior

Placeholder color should be orange when using Rust style.

### Actual behavior

Placeholder color is grey when using Rust style.

### Steps to reproduce

1. Use 2024 theme
2. Appearance > Editor > Styles. Select Rust.
3. Go to checkout.
4. Click to add order note.
5. See wrong color.

![Screenshot 2024-06-27 at 13 35 15](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/0558563b-c45c-44cb-bd6a-bccbb40dd683/87bdbed4-9362-4e63-bf35-05c6d1176df1?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi8wNTU4NTYzYi1jNDVjLTQ0Y2ItYmQ2YS1iY2NiYjQwZGQ2ODMvODdiZGJlZDQtOTM2Mi00ZTYzLWJmMzUtMDVjNmQxMTc2ZGYxIiwiaWF0IjoxNzcyMDI1NzMzLCJleHAiOjE4MDM1OTYyOTN9.3rC37qTBW0JcIrpy_jKxiNlJEEp6GbNRMnGBcurdiPo)

### WordPress Environment

WooCommerce 9.1.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/products/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
![Screenshot 2024-06-27 at 13 39 25](https://github.com/woocommerce/woocommerce/assets/90977/ba39b9f8-8cf2-4230-9313-b29c88873371)

This is also grey. @nielslange should this be orange too?

**Comment by nielslange** [comment]
Thanks for creating this issue, @mikejolley. As for the pickup address, I'd say that text should inherit the color, and thus, be displayed in orange when using the TT4 Rust style. As for the placeholder, I'm not sure if the text color should be inherited there, given it's the placeholder. @nikkivias what are your thoughts on this?

**Comment by mikejolley** [comment]
@nielslange if you clear a regular input, although its a label, it is styled like a placeholder and has orange colouring. That makes this field look odd in comparison.

**Comment by nielslange** [comment]
> @nielslange if you clear a regular input, although its a label, it is styled like a placeholder and has orange colouring. That makes this field look odd in comparison.

What do you mean by _"if you clear a regular input, although its a label, it is styled like a placeholder and has orange colouring."_? Here's what I see on my end.

Before adding text:

<img width="835" alt="Screenshot 2024-06-28 at 13 46 21" src="https://github.com/woocommerce/woocommerce/assets/3323310/a3e88a08-3b1a-48a8-835c-af9bccbf2b3a">

After adding text:

<img width="828" alt="Screenshot 2024-06-28 at 13 47 02" src="https://github.com/woocommerce/woocommerce/assets/3323310/dc5e994b-1bbc-4957-9877-982609ddf918">

After deleting text (with the field still focused):

<img width="829" alt="Screenshot 2024-06-28 at 13 47 26" src="https://github.com/woocommerce/woocommerce/assets/3323310/c1c8384b-f918-430c-bafd-1b868bc1867e">

**Comment by mikejolley** [comment]
@nielslange 

Other inputs: ![Screenshot 2024-06-28 at 12 52 25](https://github.com/woocommerce/woocommerce/assets/90977/1bb6e583-33d3-4efe-9e9f-6083543970b6)

The orange text is the label but its _styled_ like a placeholder. Contrast this to your first screenshot.

**Comment by nielslange** [comment]
Thanks for clarifying what you meant, @mikejolley.

**Comment by dominique-vias** [comment]
@nielslange I would say placeholder should be colored as is (in grey). Placeholder texts communicate their nature through color which is often, if not always grey, or at least with less contrast against the background. (Can we inherit the color and reduce transparency?)

Secondary texts can inherit the global color since they are denoted as secondary via their size.

**Comment by nielslange** [comment]
> Can we inherit the color and reduce transparency?

@nikkivias Sure, that's possible. Here's a brief demo:

```css
.wc-block-checkout__add-note .wc-block-components-textarea::placeholder {
	color: currentColor;
	opacity: 0.75;
}
``` 

**Placeholder color:**

<img width="821" alt="Screenshot 2024-07-04 at 11 57 04" src="https://github.com/woocommerce/woocommerce/assets/3323310/a7db6211-daeb-4ca7-8878-42c7d307b9a2">

**Text color:**

<img width="821" alt="Screenshot 2024-07-04 at 11 58 51" src="https://github.com/woocommerce/woocommerce/assets/3323310/173c1ee4-eed5-4bc7-ad01-3077a29efbb9">

**Comment by dominique-vias** [comment]
Thanks for the demo @nielslange Seeing it now, I think it would be better to assign a universal grey color so we don't have to worry about contrast issues. It's also a universally accepted color for placeholders.

I'm not sure what greys we have available to pull but something close to #6c757d would be good.

**Comment by nielslange** [comment]
> I'm not sure what greys we have available to pull but something close to #6c757d would be good.

In https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/css/abstracts/_colors.scss, we have the following colors defined:

```scss
// Bright colors
$discount-color: $alert-green;

$input-border-gray: #50575e;
$input-border-dark: rgba(255, 255, 255, 0.4);
$controls-border-dark: rgba(255, 255, 255, 0.6);
$input-text-active: #2b2d2f;
$input-placeholder-dark: rgba(255, 255, 255, 0.6);
$input-text-dark: #fff;
$input-background-dark: rgba(0, 0, 0, 0.1);
$select-dropdown-dark: #1e1e1e;
$select-dropdown-light: #fff;
$select-item-dark: rgba(0, 0, 0, 0.4);
$image-placeholder-border-color: #f2f2f2;

// Universal colors for use on the frontend, currently being applied to checkout blocks.
$universal-border-light: rgba(17, 17, 17, 0.11); // #e7e7e7 on white. Used for low contrast decorative elements such as section borders.
$universal-border-medium: rgba(25, 23, 17, 0.48); // Used for radio and checkbox resting state.
$universal-border-strong: rgba(17, 17, 17, 0.8); // #1e1e1e on white. Used for low contrast decorative elements such as input borders.
$universal-body-low-emphasis: rgba(17, 17, 17, 0.7); // Used for low emphasis text such as input labels.
$universal-background: rgba(17, 17, 17, 0.02); // Used for low contrast decorative elements background color.
```

On a white background, they translate to the following hex values:

- `$input-border-gray`: `#50575e`
- `$input-border-dark`: `#cccccc`
- `$controls-border-dark`: `#999999`
- `$input-text-active`: `#2b2d2f`
- `$input-placeholder-dark`: `#999999`
- `$input-text-dark`: `#ffffff`
- `$input-background-dark`: `#e6e6e6`
- `$select-dropdown-dark`: `#1e1e1e`
- `$select-dropdown-light`: `#ffffff`
- `$select-item-dark`: `#999999`
- `$image-placeholder-border-color`: `#f2f2f2`
- `$universal-border-light`: `#e7e7e7`
- `$universal-border-medium`: `#928c7d`
- `$universal-border-strong`: `#1e1e1e`
- `$universal-body-low-emphasis`: `#4d4d4d`
- `$universal-background`: `#fafafa`

Apart from that, we also consume the following two dependencies:
- [`@wordpress/base-styles/colors`](https://github.com/WordPress/gutenberg/blob/trunk/packages/base-styles/_colors.scss)
- [`@automattic/color-studio/dist/color-variables`](https://github.com/Automattic/color-studio/blob/trunk/dist/colors.json)

**Comment by yjailin** [comment]
Hello, this is a great discussion subject and i would need a bit more time to investigate the color system. Adding this to my backlog. :)

**Comment by nielslange** [comment]
> Hello, this is a great discussion subject and i would need a bit more time to investigate the color system. Adding this to my backlog. :)

If you have any questions, just let me know, @yjailin.

**Comment by tyxla** [comment]
@yoannjailin-ai what do you need in order to unblock this one?

**Comment by yoannjailin-ai** [comment]
@tyxla this issue wasn't on my tasklist for a while, getting back to it now. 🙂

**Comment by yjailin** [comment]
@tyxla hello! i have designs to fix this. Here is the [proposal](https://www.figma.com/design/fpl1YUEIW7suBFO09qymA2/Buying-Funnel-library?node-id=4821-38873&t=IP3mk6uz1VUNRbO8-1). You can see: 
(1) The idea
(2) Checkout Block Settings (colors)
(3) Checkout Block Design vs field states

The thing is, we have this [project](https://archp2.wordpress.com/2025/12/02/making-all-forms-just-work-across-the-site-as-defined-by-global-styles/) ongoing with @kkoteen and @simison (no roadmap or assignment yet) so timeline is not super clear. So i see two paths here:
(1) Fix now with provided designs above, and replace everything by shared library later.
(2) Wait for the shared fields to be available. We can still uses these designs for the fields design and color options.

@pierorocca @kkoteen @elizaan36 should we wait for the shared form fields to be built?

**Comment by kkoteen** [comment]
@yoannjailin-ai, my preference would be to make progress where we can before the shared form fields are complete, but I would be interested in @simison's perspective as well.

**Comment by simison** [comment]
There are two distinct ideas here discussed that it's best not to mix too much:
1) Shared field component between Jetpack & Woo
2) "Forms in Woo & Jetpack are using global styles
  - Including inheriting basic text input styles from GS,
  - and have lower CSS specificity for their own opinionated styles.

The former is more of a _way_ to implement this but not a goal in itself. The latter — adding global styles support — is clearer as a goal. The actual goal is that forms look consistent across the site, regardless of plugin. Anyway, this doesn't necessarily depend on using shared field components. It would be more pragmatic as a first step to get existing fields in Woo to support global styles the best we can, see where we fall short, and then consider shared components.

**Comment by kkoteen** [comment]
Appreciate the clarification! Thank you @simison

**Comment by pierorocca** [comment]
Hi @mikejolley - thanks for reporting this, and for the detail you gave.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed, or it's no longer happening on trunk.

Here's what I found: Current trunk's shared textarea component now explicitly inherits text color for the empty textarea and sets placeholder color to currentColor, which directly addresses the issue's report that checkout textarea placeholders did not inherit global text color styles. Git history identifies this as commit 2362406622d / PR #63539, 'Update checkout textarea to use global text colour for placeholder'. No open PR remains linked to the issue in the open PR metadata.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.

Fix/changelog reference: https://github.com/woocommerce/woocommerce/pull/63539
